packages/repository-json-schema/src/__tests__/integration/build-schema.integration.ts

```diff
@@ -932,5 +932,53 @@ describe('build-schema', () => {
       expect(partialSchema.required).to.equal(undefined);
       expect(partialSchema.title).to.equal('ProductPartial');
     });
+
+    context('optional properties when option "optional" is set', () => {
+      @model()
+      class Product extends Entity {
+        @property({id: true, required: true})
+        id: number;
+
+        @property({required: true})
+        name: string;
+
+        @property()
+        description: string;
+      }
+
+      it('makes one property optional when the option "optional" includes one property', () => {
+        const originalSchema = getJsonSchema(Product);
```

Comment:
@nabdelgadir A question for the generic type `T` in this PR: if you don't specify the type when calling `getJsonSchema(ModelCtor)`, then where is it provided?

Reply:
I think TS 3.5.x defaults to `unknown`.